internal/fs/inode/dir.go

```diff
@@ -835,20 +835,43 @@ func (d *dirInode) CreateChildSymlink(ctx context.Context, name string, target s
 
 // LOCKS_REQUIRED(d)
 func (d *dirInode) CreateChildDir(ctx context.Context, name string) (*Core, error) {
+	// Generate the full name for the new directory.
 	fullName := NewDirName(d.Name(), name)
-	o, err := d.createNewObject(ctx, fullName, nil)
-	if err != nil {
-		return nil, err
+	var core *Core
+
+	// Check the bucket type.
+	if d.isBucketHierarchical() {
+		// For hierarchical buckets, create a folder.
+		f, err := d.bucket.CreateFolder(ctx, fullName.objectName)
+		if err != nil {
+			return nil, err
+		}
+		// Convert the folder to a minimal object.
+		m := f.ConvertFolderToMinObject()
+		core = &Core{
+			FullName:  fullName,
+			Bucket:    d.Bucket(),
+			MinObject: m,
+		}
+	} else {
+		// For non-hierarchical buckets, create a new object.
+		o, err := d.createNewObject(ctx, fullName, nil)
+		if err != nil {
+			return nil, err
+		}
+		// Convert the object to a minimal object.
+		m := storageutil.ConvertObjToMinObject(o)
+		core = &Core{
+			Bucket:    d.Bucket(),
```

Comment:
This is also common for both if-else. can you move it out of if-else

Reply:
Done.